### ISSUE TITLE
Search button in docs missing aria-haspopup

### ISSUE THREAD

**Issue by amberhinds** [LibreChat-AI/librechat.ai#514]
The search button in the docs is missing a warning for screen reader users that it will open a modal. It also announces as a group, which may be confusing for screen reader users.

This can be seen at https://www.librechat.ai/docs

<img width="1106" height="485" alt="Image" src="https://github.com/user-attachments/assets/61029766-51c9-48de-9423-9cab47f00a0c" />

Here's the code for the button:
```
<button type="button" data-search-full="" class="inline-flex items-center gap-2 border bg-fd-secondary/50 p-1.5 text-sm text-fd-muted-foreground transition-colors hover:bg-fd-accent hover:text-fd-accent-foreground rounded-lg max-md:hidden"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-search ms-1 size-4"><circle cx="11" cy="11" r="8"></circle><path d="m21 21-4.3-4.3"></path></svg>Search<div class="ms-auto inline-flex gap-0.5"><kbd class="rounded-md border bg-fd-background px-1.5">⌘</kbd><kbd class="rounded-md border bg-fd-background px-1.5">K</kbd></div></button>
``` 

The group announcement is caused by the unlabelled, but not hidden SVG and multiple divs in the button.

To fix this:
1. Add `aria-haspopup="dialog"` to the button.
2. Add an `aria-label` to the button with the visible text "Search ⌘ K" - then screen reader will only read the name and not announce it as a group.
